javascripts/discourse/components/user-color-palette-selector.gjs

```diff
@@ -0,0 +1,175 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import AsyncContent from "discourse/components/async-content";
+import icon from "discourse/helpers/d-icon";
+import { reload } from "discourse/helpers/page-reloader";
+import {
+  listColorSchemes,
+  loadColorSchemeStylesheet,
+  updateColorSchemeCookie,
+} from "discourse/lib/color-scheme-picker";
+import { i18n } from "discourse-i18n";
+import DMenu from "float-kit/components/d-menu";
+import UserColorPaletteMenuItem from "./user-color-palette-menu-item";
+
+const HORIZON_PALETTES = [
+  "Horizon",
+  "Marigold",
+  "Violet",
+  "Lily",
+  "Clover",
+  "Royal",
+];
+
+export default class UserColorPaletteSelector extends Component {
+  @service currentUser;
+  @service keyValueStore;
+  @service site;
+  @service session;
+  @service interfaceColor;
+
+  constructor() {
+    super(...arguments);
+
+    // Only need to do this for anon because the current user will have their
+    // color scheme cookie set and other automatic things from core loads their
+    // preference.
+    if (!this.currentUser && this.selectedColorPaletteId) {
```

Comment:
This is not working if the system preference is set to dark.

The default palette is used always.

It works fine when the system preference is set to light

Reply:
The system preference stuff is now working, but the interface color switcher does not work for anons, see /t/149830/18 internally. This stuff just feels so overly complicated...